### PR TITLE
Add new keyboard "Rookey"

### PR DESCRIPTION
**Pull request by takashicompany**
## Description

Thank you for everything.

I have firmware for the new keyboard and would appreciate your review.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/takashicompany/rookey/config.h
Comment (waffle87) [inline keyboards/takashicompany/rookey/config.h:1]: This file should be removed.

File: keyboards/takashicompany/rookey/info.json [hunk 1]
```diff
@@ -0,0 +1,115 @@
+{
+    "manufacturer": "takashicompany",
+    "keyboard_name": "Rookey",
+    "maintainer": "takashicompany",
+    "bootloader": "caterina",
```
Comment (waffle87) [inline keyboards/takashicompany/rookey/info.json:5]:
```suggestion
```

File: keyboards/takashicompany/rookey/info.json [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
+    },
+    "matrix_pins": {
+        "direct": [
+            ["F4", "F5", "F6", "F7", "B1", "B3", "B2", "B6", "B5"]
+        ]
+    },
+    "processor": "atmega32u4",
```
Comment (waffle87) [inline keyboards/takashicompany/rookey/info.json:30]:
```suggestion
```

File: keyboards/takashicompany/rookey/info.json
Comment (waffle87) [inline keyboards/takashicompany/rookey/info.json:1]: Please run `qmk format-json` on this file.

File: keyboards/takashicompany/rookey/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,25 @@
+// Copyright 2023 QMK
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+#include "quantum.h"
```
Comment (waffle87) [inline keyboards/takashicompany/rookey/keymaps/default/keymap.c:5]:
```suggestion
```

File: keyboards/takashicompany/rookey/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
+        LT(1, KC_Q), KC_W, KC_E, KC_R,
+        KC_A, KC_S, KC_D, KC_F,
+        KC_V
+    ),
+    [1] = LAYOUT(
+        LT(1, KC_TRNS), KC_TRNS, KC_TRNS, KC_TRNS,
```
Comment (waffle87) [inline keyboards/takashicompany/rookey/keymaps/default/keymap.c:14]:
```suggestion
        _______, KC_TRNS, KC_TRNS, KC_TRNS,
```

File: keyboards/takashicompany/rookey/keymaps/via/config.h
Comment (waffle87) [inline keyboards/takashicompany/rookey/keymaps/via/config.h:1]: This file should be removed. The content can be moved to `info.json`, if you have a good reason for needing 32 layers.

File: keyboards/takashicompany/rookey/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+    [30] = LAYOUT(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS
+    ),
+
+    [31] = LAYOUT(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS
```
Comment (waffle87) [inline keyboards/takashicompany/rookey/keymaps/via/keymap.c:199]:
```suggestion
```

File: keyboards/takashicompany/rookey/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+    [23]    =   { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [24]    =   { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [25]    =   { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [26]    =   { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [27]    =   { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [28]    =   { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [29]    =   { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [30]    =   { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [31]    =   { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) }
+
```
Comment (waffle87) [inline keyboards/takashicompany/rookey/keymaps/via/keymap.c:238]:
```suggestion

    [0]     =   { ENCODER_CCW_CW(KC_MS_WH_UP, KC_MS_WH_DOWN) }
```

File: keyboards/takashicompany/rookey/rules.mk [hunk 7]
```diff
@@ -0,0 +1,2 @@
+# This file intentionally left blank
+ENCODER_MAP_ENABLE = yes
```
Comment (waffle87) [inline keyboards/takashicompany/rookey/rules.mk:2]: This cannot be enabled at the keyboard level.
```suggestion
```

File: keyboards/takashicompany/rookey/readme.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
+# Rookey
+
+![takashicompany/rookey](https://i.imgur.com/kLQDeZZh.jpg)
+
+*A short description of the keyboard/project*
```
Comment (waffle87) [inline keyboards/takashicompany/rookey/readme.md:5]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/takashicompany/rookey/keymaps/via/config.h
Comment (waffle87) [inline keyboards/takashicompany/rookey/keymaps/via/config.h:1]: This file should be removed.

File: keyboards/takashicompany/rookey/info.json [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        "direct": [
+            ["F4", "F5", "F6", "F7", "B1", "B3", "B2", "B6", "B5"]
+        ]
+    },
+    "url": "https://github.com/takashicompany/rookey",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x0056",
+        "vid": "0x7463"
+    },
```
Comment (waffle87) [inline keyboards/takashicompany/rookey/info.json:31]:
```suggestion
    },
    "dynamic_keymap": {
        "layer_count": 32
    },
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/takashicompany/rookey/info.json (hunk 9 above)
Comment (drashna) [reply keyboards/takashicompany/rookey/info.json:31]: I don't know. 16 is more reasonable.  And 8 would be more than plenty.

**Review by drashna — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**